lib/ast/AST.cpp

```diff
@@ -1829,6 +1831,72 @@ void KOREDefinition::print(std::ostream &Out, unsigned indent) const {
   }
 }
 
+// Binary serialization
+
+void KOREVariablePattern::serialize_to(serializer &s) const {
+  sort->serialize_to(s);
+  s.emit(header_byte<KOREVariablePattern>);
+  name->serialize_to(s);
+}
+
+void KORECompositePattern::serialize_to(serializer &s) const {
+  for (auto const &arg : arguments) {
+    arg->serialize_to(s);
+  }
+
+  constructor->serialize_to(s);
+
+  s.emit(header_byte<KORECompositePattern>);
+  s.emit(int16_t(arguments.size()));
```

Comment:
I would like a graceful failure in case we have a number of arguments that cannot fit in an `int16_t`.